functions.php

```diff
@@ -1,8 +1,7 @@
 <?php
 
-function view($page, $data = []) {
-    extract($data);
-
+function view($page, $errors = []) {
```

Comment:
i choosed `$templateVars` as you suggested ... 
So when throwing exception i add the error 
`$templateVars['error'] = $e->getMessage() `
but in `HomeController` i find it better  to `compact` different variables
 `view('index', compact('repos', 'paginator', 'args')`  
it works but is this a wrong way to handle this ? 
should i pass theses variables to `$templateVars` and stick with 
`view('404', $templateVars)` ? 
like `$templateVars['repos']` `$templateVars['paginator']` ? 
which approach is better ?

Reply:
Definitely the second approach is preferred and also more readable, no "magic" is happening there. 

Compare your `view('404', compact('repos', 'paginator', 'args')` with this:

```php
view('404', [
  'repos' => $repos,
  'paginator' => $paginator,
  'args' => $args,
]);
```

Or, even this, because some variables were introduced just to be used in `compact()`:

```php
view('404', [
  'repos' => $repos,
  'paginator' => new Paginator,
  'args' => $this->getArgs(),
]);
```

Reply:
@vladimino Done, maybe we should now merge this branch now and later create a separate issue for testing ?